src/runtime/react/render.js

```diff
@@ -1,14 +1,29 @@
 import { runEffects } from "../effects";
 import renderInternal from "../renderInternal";
+import replaceNode, { toArray } from "../replaceNode";
 
 const map = new WeakMap();
-export default (element, parentNode) => {
+export default (element, parentNode, callback) => {
 	let context = map.get(parentNode);
 	if (!context) {
 		map.set(parentNode, (context = { $$: { node: parentNode } }));
 	}
 	renderInternal(context, element, "node", false);
-	const node = context.node;
-	if (node.parentElement !== parentNode) parentNode.appendChild(node);
+	let node = context.node;
+	if (Array.isArray(node)) {
+		if (parentNode.childNodes.length > 0) {
+			replaceNode(parentNode.childNodes, node);
+		} else {
```

Comment:
alternative is making a more general purpose method used by replaceNode or by this.